api/src/services/dataset.js

```diff
@@ -480,6 +477,34 @@ async function add_files({ dataset_id, data }) {
   });
 }
 
+/**
+ * Creates a new dataset within a transaction.
+ *
+ * @param {Object} tx - Database transaction manager.
+ * @param {Object} data - The data object containing details of the dataset to be created.
+ * @return {Promise<Object|undefined>} Returns the created dataset object if successfully created, otherwise returns undefined if a dataset with the same name and type already exists.
+ */
+async function createDatasetInTransaction(tx, data) {
```

Comment:
This method allows us to continue creating datasets using the existing logic (method `createDataset()`, while also allowing other services to perform other database operations within a transaction by providing their own Transaction Manager instance.